maintainers/scripts/haskell/transitive-broken-packages.nix

```diff
@@ -0,0 +1,21 @@
+let
+  nixpkgs = import ../../..;
+  inherit (nixpkgs {}) pkgs lib;
+  getEvaluating = x:
+    builtins.filter (x: !isNull x) (
+      lib.mapAttrsToList (
+        k: v: if (builtins.tryEval (v.outPath or null)).success && lib.isDerivation v && !v.meta.broken then k else null
+      ) x
+    );
+  brokenDeps = lib.subtractLists
+    (getEvaluating pkgs.haskellPackages)
+    (getEvaluating (nixpkgs { config.allowBroken = true; }).haskellPackages);
```

Comment:
You can implement this much simpler:

```nix
let
  brokenDeps = builtins.attrNames (
    lib.filterAttrs (
      _: v: !((builtins.tryEval (v.outPath or null)).success)
         && lib.isDerivation v
         && !v.meta.broken
    ) haskellPackages
```

The approach is to filter out all packages that fail to evaluate *despite* not being marked as broken.

Reply:
Sadly I don‘t think that will work. That would filter out everything that fails to eval.
That could be anything from missing specific haskell-lib-versions to asserts in our code.

Reply:
We don't have any asserts in `haskellPackages`, missing attributes is not a catchable eval error in any case so both our codes would fail to eval themselves.

Reply:
As of now there are three `assert` in configuration-common.nix.

But the point about not catchable eval errors is a good one.